### PR TITLE
#28409: Add condition to force witing to disk

### PR DESCRIPTION
**Pull request by domoratskii**
Check pls

### CODE REVIEW

**Comment by domoratskii** [comment]
@shifteee check pls. Is it okay ?

**Review by AnDeVerin — APPROVED** [review]
LGTM

File: lib/stats-writer-plugin.js [hunk 1]
```diff
@@ -89,11 +91,16 @@ class StatsWriterPlugin {
...
+      if (IS_FIRST_START_DEVSERVER) {
+        let file = path.join(curCompiler.compiler.options.output.path, this.opts.filename);
+
+        IS_FIRST_START_DEVSERVER = false;
+        fs.writeFile(file, statsStr, callback);
+      }
+
 
-      return;
+      return callback && callback();
```
Comment (shifteee) [inline lib/stats-writer-plugin.js:103]: 2nd call?

**Review by shifteee — COMMENTED** [review]
it is not clear what purpose of this flag

File: lib/stats-writer-plugin.js (hunk 1 above)
Comment (domoratskii) [reply lib/stats-writer-plugin.js:103]: A simple `callback()`will suffice

**Review by domoratskii — COMMENTED** [review]
> it is not clear what purpose of this flag

This is to ensure that the webpack-stats-plugin doesn't write `.hot-update.js` when we use webpack-dev-server and it's good for optimization

File: lib/stats-writer-plugin.js [hunk 2]
```diff
@@ -6,6 +6,8 @@ const path = require("path");
 const INDENT = 2;
 const DEFAULT_TRANSFORM = (data) => JSON.stringify(data, null, INDENT);
 
+let IS_FIRST_START_DEVSERVER = true;
```
Comment (podkot) [inline lib/stats-writer-plugin.js:9]: maybe, `IS_FIRST_RUN_WO_HMR`?

Comment (domoratskii) [reply lib/stats-writer-plugin.js:9]: I think that `IS_FIRST_START_DEVSERVER` is okay because it's not only for `HMR`

**Comment by podkot** [comment]
@domoratskii bump minor pls

**Review by podkot — APPROVED** [review]

**Pull request merged**